### PR TITLE
Stand alone bug logging

### PR DESCRIPTION
**Pull request by arvindsingh-cloud**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+					}
+					else {
+						Write-Host "Scan result file is not in correct format. Please remove [Tags] column if it is there in the file." -ForegroundColor Red
+						return;
+					}
+				}
+			}
+				$resolver = [Resolver]::new($Organization, $PATToken);
+
+				$blPnAcc = $false;
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:234]: buglogprojectaccess

Comment (arvindsingh-cloud) [reply src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:234]: comment

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	}
+	
+}
+
+function ValidateBugTemplate {
+	param (
+		$BugTemplate
+	)
+
+	$mandetorytemplateItems = @("/fields/System.Title", "/fields/System.AreaPath", "/fields/System.IterationPath", "/fields/System.Tags", "/fields/System.AssignedTo")
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:297]: spelling

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+	try {
+		foreach ($templateItem in $mandetorytemplateItems) {
+			if($templateItem -notin $BugTemplate.path) {
+				Write-Host "Bug template formate is not correct. Mandetory field is not supplied in the template." -ForegroundColor Red
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:301]: format and mandatory fields are

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+				return $false;
+			}	
+		}
+		
+		return $true;
+	}
+	catch {
+		Write-Host "Could not parse bug template. please validate the template formate." -ForegroundColor Red
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:309]: format

File: src/AzSK.ADO/Framework/Abstracts/CommandBase.ps1 [hunk 5]
```diff
@@ -182,7 +182,10 @@ class CommandBase: AzSKRoot {
 			[PartialScanManager]::ControlResultsWithSARIFSummary=@()
         }
 		# Publish command complete events
-        $this.CommandCompleted($methodResult);
+		if(!$this.InvocationContext.BoundParameters["ScanResultFilePath"])
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Abstracts/CommandBase.ps1:185]: comment for the condition

File: src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1 [hunk 6]
```diff
@@ -753,6 +829,12 @@ class AutoBugLog {
             }
             $SecuritySeverity = $this.GetSecuritySeverity($secSeverity)		
         }
+        elseif ($BugTemplateInCMD) {
+         $BugTemplate = $BugTemplateInCMD;  
+         if ($BugTemplate.length -gt 7 -and $BugTemplate[7].path -like "*HowFound*") { #get it outside.
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1:834]: comment

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+			[ConfigurationHelper]::PolicyCacheContent = @()
+			[ConfigurationHelper]::OnlinePolicyEnabled = $false
+			[ConfigurationHelper]::OssPolicyUrl = ""
+			[ConfigurationHelper]::OssPolicyEnabled = $false
+			[ConfigurationHelper]::LocalPolicyEnabled = $false
+			[ConfigurationHelper]::ConfigVersion = ""
+			[AzSKSettings]::Instance = $null
+			[AzSKConfig]::Instance = $null
+			[ConfigurationHelper]::ServerConfigMetadata = $null
+			#Refresh singlton in different commands. (Powershell session keep cach object of the class, so need to make it null befor command run)
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:124]: singleton

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+function ValidateBugTemplate {
+	param (
+		$BugTemplate
+	)
+
+	$mandatorytemplateItems = @("/fields/System.Title", "/fields/System.AreaPath", "/fields/System.IterationPath", "/fields/System.Tags", "/fields/System.AssignedTo")
+	try {
+		foreach ($templateItem in $mandatorytemplateItems) {
+			if($templateItem -notin $BugTemplate.path) {
+				Write-Host "Bug template format is not correct. mandatory fields are not supplied in the template." -ForegroundColor Red
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:302]: Mandatory (caps)

File: src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1 [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+				return $false;
+			}	
+		}
+		
+		return $true;
+	}
+	catch {
+		Write-Host "Could not parse bug template. please validate the template format." -ForegroundColor Red
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AutoBugLogCMD/AutoBugLogSVT.ps1:310]: Please(Caps)

**Review by Garima-msft — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**